go/vt/vtgate/executor_select_test.go

```diff
@@ -113,6 +114,96 @@ func TestSelectDBA(t *testing.T) {
 	utils.MustMatch(t, wantQueries, sbc1.Queries)
 }
 
+func TestSystemVariablesUsingSetVarComment(t *testing.T) {
+	executor, sbc1, _, _ := createExecutorEnv()
+	executor.normalize = true
+
+	session := NewSafeSession(&vtgatepb.Session{TargetString: "TestExecutor", SystemVariables: map[string]string{"sql_mode": "only_full_group_by"}})
+
+	query := "select 1 from information_schema.table"
+	_, err := executor.Execute(context.Background(), "TestSelect", session, query, map[string]*querypb.BindVariable{})
+	require.NoError(t, err)
+	require.False(t, session.InReservedConn())
+	for _, shardSession := range session.ShardSessions {
+		require.EqualValues(t, 0, shardSession.ReservedId)
+	}
+
+	wantQueries := []*querypb.BoundQuery{
+		{Sql: "select /*+ SET_VAR(sql_mode = only_full_group_by) */ :vtg1 from information_schema.`table`", BindVariables: map[string]*querypb.BindVariable{"vtg1": {Type: sqltypes.Int64, Value: []byte("1")}}},
+	}
+
+	utils.MustMatch(t, wantQueries, sbc1.Queries)
```

Comment:
after this check.
we should execute another set sys_var with sysvar that does not support SET_VAR. and check the session again.
then execute a select query and see that both the setting, previous one and the new one are applied and reserved connection is created.

Reply:
done via 23c2f6a